### DISCUSSION TITLE
Show rule on a section reference

### DISCUSSION THREAD

**Discussion by CDucloux** [typst/typst#4427] [Q&A] [answered]
I would like to make sure that all my section references in a typst document are underlined. I tried to create a show rule to do it but unable to make it work. It would look like this . The goal is to make the @section look like what I did above with the `underline` func

```typst
= Problématiques du stage <section>

#underline(
  stroke: 1.5pt + blue_edf,
  offset: 3pt,
  ref(supplement: "Section", <section>)
)

@section
```
<img width="313" alt="test" src="https://github.com/typst/typst/assets/122461892/29d15ddf-daf9-4fbb-8dda-3e079d9e99e2">

**Comment by Mc-Zen** [comment, marked as answer]
Hi, you can do
```typ

#show ref: it => {
  let el = it.element
  if el != none and el.func() == heading and el.level == 1 {
    underline(stroke: 1.5pt + blue, offset: 3pt, it)
  } else { it }
}
```
We create a `show` rule for `ref` that filters for the `element` field of the reference. This `element` may be `none` (for citations for example). Then we check that the element is indeed a heading and that the level is `1`. Alternatively, the last condition could be `el.supplement == [Section]`, however, this does not work across different languages. 

Now you can just reference your sections via `@section` and they will automatically have the desired style.

**Comment by CDucloux** [comment]
I adapted your show rule and it is almost perfect !

```typst
#show ref: it => {
  let el = it.element
  if el != none and el.func() == heading {
    underline(stroke: 1.5pt + blue, offset: 3pt, it)
  } else { it }
}
```

The only issue is that instead of "Section", "Chapitre" is displayed. I don't know if there is a way to re-parametrize the default values when setting the language, because this is clearly the issue there. 
<img width="68" alt="test" src="https://github.com/typst/typst/assets/122461892/ea512a18-27a5-4df1-a1ac-f270f4ffdf9a">

**Reply by Mc-Zen** [reply]
That's because you've set the language to French and that's the default for it. You can change this so-called supplement with
```typ
#set heading(supplement: [Section])
```
